src/main/java/fr/pilato/elasticsearch/crawler/fs/client/ElasticsearchClientManager.java

```diff
@@ -100,12 +100,13 @@ public void createIndexAndMappings(FsSettings settings, boolean updateMapping) t
 
         try {
             // If needed, we create the new mapping for files
-            if (!settings.getFs().isJsonSupport() && !settings.getFs().isXmlSupport()) {
+            if (settings.getFs().isAddAsInnerObject() == false ||  (!settings.getFs().isJsonSupport() && !settings.getFs().isXmlSupport())) {
```

Comment:
haha! That's definitely how elasticsearch policy wants negative tests to be implemented :)
You can keep `== false` or use `!` as you prefer.